src/HexagonAlignLoads.cpp

```diff
@@ -0,0 +1,242 @@
+#include "HexagonAlignLoads.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+#include "Scope.h"
+#include "Bounds.h"
+#include "ModulusRemainder.h"
+#include "Simplify.h"
+namespace Halide {
+namespace Internal {
+using std::vector;
+
+class HexagonAlignLoads : public IRMutator {
+public:
+    HexagonAlignLoads(Target t, int vs) : target(t), vector_size(vs) {}
+private:
+    Target target;
+    enum AlignCheck { Aligned, Unaligned, NoResult };
+    // size of a vector in bytes.
+    int vector_size;
+    /** Alignment info for Int(32) variables in scope. */
```

Comment:
This is a definition file, I think we should not use a doxygen comment here. I know some other stuff does this, but we should at least not introduce a new one.